sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ConstantColumnVector.java

```diff
@@ -0,0 +1,212 @@
+package org.apache.spark.sql.execution.vectorized;
+
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.sql.vectorized.ColumnVector;
+import org.apache.spark.sql.vectorized.ColumnarArray;
+import org.apache.spark.sql.vectorized.ColumnarMap;
+import org.apache.spark.unsafe.types.UTF8String;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * This class adds the constant support to ColumnVector.
+ * It supports all the types and contains put APIs,
+ * which will put the exact same value to all rows.
+ *
+ * Capacity: The vector only stores one copy of the data, and acts as an unbounded vector
+ * (get from any row will return the same value)
+ */
+public class ConstantColumnVector extends ColumnVector {
+
+  private byte nullData;
+  private byte byteData;
+  private short shortData;
+  private int intData;
+  private long longData;
+  private float floatData;
+  private double doubleData;
+  private byte[] byteArrayData;
+  private ConstantColumnVector childData;
+  private ColumnarArray arrayData;
+  private ColumnarMap mapData;
+
+  /**
+   * Sets up the data type of this constant column vector.
+   * @param type
+   */
+  public ConstantColumnVector(DataType type) {
+    super(type);
+  }
+
+  @Override
+  public void close() {
+    byteArrayData = null;
+    childData = null;
+    arrayData = null;
+    mapData = null;
+  }
+
+  @Override
+  public boolean hasNull() {
+    return nullData == 1;
+  }
+
+  @Override
+  public int numNulls() {
+    return -1;
```

Comment:
why `-1` here?

Reply:
this `ConstantColumnVector` is 'boundless', all values are the same, and no capacity (no total number of rows), I am also wondering what we should return here - maybe something like an `UnimplementedException`? cc: @cloud-fan

Reply:
we can know the number of rows from the context, right? the `numNulls` should either be 0 or `numRows`.